ports/psoc6/drivers/psoc6_gpio.c

```diff
@@ -0,0 +1,105 @@
+/* CYHAL GPIO functions */
+
+#include "psoc6_gpio.h"
+
+gpio_init_rslt gpio_init(cyhal_gpio_t pin, cyhal_gpio_direction_t direction, cyhal_gpio_drive_mode_t drive_mode, bool init_val) {
+    return cyhal_gpio_init(pin, direction, drive_mode, init_val);
+}
+
+// function to get HSIOM config of a pin
+en_hsiom_sel_t pin_get_hsiom_func(uint32_t pin) {
+    return Cy_GPIO_GetHSIOM(CYHAL_GET_PORTADDR(pin), CYHAL_GET_PIN(pin));
+}
+
+// function to get cypdl drive modes, correlated to cyhal drive modes in file: cyhal_gpio.c
+// Note: drive modes are enumed in cy_gpio.h and are also distinguised for pins with input buffer on or off (configured as input or output respectively)
+uint32_t gpio_get_drive(uint32_t pin) {
+    return Cy_GPIO_GetDrivemode(CYHAL_GET_PORTADDR(pin), CYHAL_GET_PIN(pin));
+}
+
+// function to check if pin is in mode Pin.OPEN_DRAIN.
+// drive comparisons done with PDL drive modes since function is from PDL (not HAL)
+bool gpio_is_open_drain(uint32_t pin) {
+    if (gpio_get_drive(pin) == CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW) {
+        return 1;
```

Comment:
Please change all occurences of 0/1 by the respective value of true / false. That is the real purpose of using stdbool.h, in the end.

Reply:
Done as suggested.